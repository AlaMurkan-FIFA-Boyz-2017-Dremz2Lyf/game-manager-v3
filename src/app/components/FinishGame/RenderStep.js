import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { Header, Segment, Table } from 'semantic-ui-react';
import ImageReader from './ImageReader';
import Directions from './Directions';
import TwitterSearch from './TwitterSearch';
import Searching from './Searching';

const TRACKED_STATS = [
  'score',
  'shots',
  'shots on target',
  'possession %',
  'tackles',
  'fouls',
  'yellow cards',
  'red cards',
  'injuries',
  'offsides',
  'corners',
  'shot accuracy %',
  'pass accuracy %',
];

const Verify = ({ results }) => (
  <Segment basic>
    <Header color="teal">
      Verify
    </Header>
    <p>Please verify that the results match the image below.</p>
    <Table compact inverted unstackable>
      <Table.Header>
        <Table.Row>
          <Table.HeaderCell textAlign="left">Home</Table.HeaderCell>
          <Table.HeaderCell textAlign="center">Stat</Table.HeaderCell>
          <Table.HeaderCell textAlign="right">Away</Table.HeaderCell>
        </Table.Row>
      </Table.Header>

      <Table.Body>
        {TRACKED_STATS.map((statName) => {
          const home = results.home[statName];
          const away = results.away[statName];
          return (
            <Table.Row key={statName}>
              <Table.Cell textAlign="left">{home}</Table.Cell>
              <Table.Cell textAlign="center">{statName}</Table.Cell>
              <Table.Cell textAlign="right">{away}</Table.Cell>
            </Table.Row>
          );
        })}
      </Table.Body>
    </Table>
  </Segment>
);

Verify.propTypes = {
  results: PropTypes.object,
};


class RenderStep extends PureComponent {
  static propTypes = {
    step: PropTypes.number,
  }

  getStepComponent = (step) => {
    const stepMap = [
      Directions,
      TwitterSearch,
      Searching,
      ImageReader,
      Verify,
    ];

    return stepMap[step];
  }

  render() {
    const { step } = this.props;

    const CurrentStep = this.getStepComponent(step);
    return (
      <CurrentStep {...this.props} />
    );
  }
}

export default RenderStep;
