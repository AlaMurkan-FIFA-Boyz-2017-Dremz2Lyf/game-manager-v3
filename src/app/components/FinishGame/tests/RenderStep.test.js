import React from 'react';
import { shallow } from 'enzyme';

import RenderStep from '../RenderStep';

describe('RenderStep', () => {
  it('should match the snapshot', () => {
    const wrapper = shallow(
      <RenderStep step={0} />,
    );

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('Directions').length).toBe(1);
  });
});
