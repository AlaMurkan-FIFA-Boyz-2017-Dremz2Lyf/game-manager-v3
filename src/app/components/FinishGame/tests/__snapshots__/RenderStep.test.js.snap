// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RenderStep should match the snapshot 1`] = `"<div class=\\"ui basic center aligned segment\\"><h2 class=\\"ui teal header\\"><i aria-hidden=\\"true\\" class=\\"share icon\\"></i><div class=\\"content\\">Directions</div></h2><div class=\\"ui teal message\\">Please select Match Facts in the post-match menu. When there, please use the share feature to take a screenshot.</div></div>"`;
